### ISSUE TITLE
Too many log messages after upgrade to pipx 1.4.0 on MacOS

### ISSUE THREAD

**Issue by sorokine** [pypa/pipx#1182]
**Describe the bug**

After upgrade to version 1.4.0 pipx produces lots of debugging output that is not needed for normal use.

**How to reproduce**

```zsh
> pipx list
pipx version is 1.4.0
Default python interpreter is '/usr/local/opt/python@3.12/libexec/bin/python'
running <checking pip's availability>
Time since last upgrade of shared libs, in seconds: 30838. Upgrade will be run by pipx if greater than 2592000.
running <checking pip's availability>
running <checking pip's availability>
running <checking pip's availability>
running <checking pip's availability>
running <checking pip's availability>
running <checking pip's availability>
...
```

**Environment**

```zsh
> uname -a
Darwin ALEXMAC 23.2.0 Darwin Kernel Version 23.2.0: Wed Nov 15 21:54:10 PST 2023; root:xnu-10002.61.3~2/RELEASE_X86_64 x86_64
> env | grep '^PIP_'
## no output ##
> pipx --version
1.4.0
> brew info pipx
==> pipx: stable 1.4.0 (bottled), HEAD
Execute binaries from Python packages in isolated environments
https://pipx.pypa.io
/usr/local/Cellar/pipx/1.4.0 (144 files, 975.8KB) *
  Poured from bottle using the formulae.brew.sh API on 2024-01-01 at 12:20:10
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/p/pipx.rb
License: MIT
==> Dependencies
Required: python-argcomplete ✔, python-packaging ✔, python@3.12 ✔
==> Options
--HEAD
	Install HEAD version
==> Caveats
zsh completions have been installed to:
  /usr/local/share/zsh/site-functions
==> Analytics
install: 16,037 (30 days), 34,318 (90 days), 60,440 (365 days)
install-on-request: 16,018 (30 days), 34,256 (90 days), 60,327 (365 days)
build-error: 1 (30 days)
> echo $ZSH_VERSION
5.9
```

**Expected behavior**

No superfluos output.

**Comment by sorokine** [comment]
The source of the messages about pip is likely this line: https://github.com/pypa/pipx/blob/main/src/pipx/shared_libs.py#L59

**Comment by ealap** [comment]
I came here to say that the issue also exists for installations on Linux and later found that this is now fixed at #1178

**Comment by sorokine** [comment]
Thanks, I am closing the issue as a duplicate of #1178
